src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,84 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import React, { useState } from 'react';
+import { Select } from '../../types/Select';
+
+type Props = {
+  onChangeQuery: (query: string) => void,
+  onChangeSelect: (event: string) => void,
+  query: string,
+
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  onChangeQuery = () => {},
+  onChangeSelect = () => {},
+  query,
+}) => {
+  const [selectOption, setSelectOption] = useState<string>();
```

Comment:
Why you need this state here, you can use parent state, the same as you did it with `query`

Reply:
> Why you need this state here, you can use parent state, the same as you did it with `query`

1) I have the same question🤷‍♂️ 

2) Update handlers names in all components, here as well `onChangeSetQuery`, `onChangeOption`.

3) Use one style for your handlers, don't combine function expressions and function declarations like this:
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/63243136/fff587f0-e328-42ca-b2c9-94933a7be7a4)